### PR TITLE
Connect4 game

### LINKED ISSUE #125: Connect 4 (C4) game with Python
# Description

Create a GUI based Connect 4 game with Python

## Type of issue

- [x] Feature (New Script)

## Checklist:

- [x] I have read the project guidelines.
- [x] I have checked all the existing projects, before submitting a new project issue.
- [x] I have checked previous issues to avoid duplicates.
- [x] This issue will be meaningful for the project.

### PR DESCRIPTION
**Pull request by SuvamPrasd**
# Description

This is a connect 4 games developed using a python programming language. For this program, I have used the pygame package and Tkinter package for GUI.

Fixes #125 

Replace `issue_no` in the above line, with the issue related to this PR.

## Type of change

Choosing one or more options from the following as per the nature of your Pull request.
- NOTE: *These boxes can be checked using **`[X]`***

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation Update

# Checklist:
Please tick all the boxes that are fulfilled by your Pull Request.

- [x] I have named my files and folder, according to this project's guidelines.
- [x] My code follows the style guidelines of this project.
- [x] My Pull Request has a descriptive title. (not a vague title like `Update index.md`)
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have created a helpful and easy to understand `README.md`, according to the given [`README_TEMPLATE.`](https://github.com/Python-World/Python_and_the_Web/blob/master/README_TEMPLATE.md)
- [x] I have included a requirements.txt file (if external libraries are required.)
- [x] My changes do not produce any warnings.
- [x] I have added a working sample/screenshot of the script.

### CODE REVIEW

File: Scripts/Miscellaneous/Connect4Game/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Before starting the game make sure to install pygame package into your local system or virtual environment
```
Comment (AdityaJ7) [inline Scripts/Miscellaneous/Connect4Game/requirements.txt:1]: requirements.txt file must include only this:

numpy==1.19.2
pygame==1.9.6

File: Scripts/Miscellaneous/Connect4Game/connect4game.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+def is_valid_location(board, col):
+	return board[ROW_COUNT-1][col] == 0
+
+def get_next_open_row(board, col):
+	for r in range(ROW_COUNT):
+		if board[r][col] == 0:
+			return r
+
+# def print_board(board):
```
Comment (AdityaJ7) [inline Scripts/Miscellaneous/Connect4Game/connect4game.py:39]: Remove this commented code

**Review by AdityaJ7 — CHANGES_REQUESTED** [review]

**Review by tusharnankani — CHANGES_REQUESTED** [review]
https://screenrec.com/share/A6OIFtuWSr

- Please change this behavior, once the column is full, the color still changes, leading to a wrong behaviour!

Please look into the matter!

File: Scripts/Miscellaneous/Connect4Game/requirements.txt (hunk 1 above)
Comment (SuvamPrasd) [reply Scripts/Miscellaneous/Connect4Game/requirements.txt]: Updated the requirements.txt file as per as you requested.

File: Scripts/Miscellaneous/Connect4Game/connect4game.py (hunk 2 above)
Comment (SuvamPrasd) [reply Scripts/Miscellaneous/Connect4Game/connect4game.py]: The commented code has been removed from the source code.

**Comment by AdityaJ7** [comment]
@all-contributors please add @SuvamPrasd for code

**Comment by tusharnankani** [comment]
Good work! @SuvamPrasd 

Expecting more contributions from your end! 👍🏻

**Comment by SuvamPrasd** [comment]
Yeah sure dude 😊

**Pull request merged**